.github/workflows/build-and-publish-container-images.yml

```
@@ -17,7 +17,7 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
       - name: Log in to the Container registry
-        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        uses: docker/login-action@3d58c274f17dffee475a5520cbe67f0a882c4dbb
```

```suggestion
        uses: docker/login-action@v3
```
I would suggest using version numbers instead of hashes. The official examples also use it that way. See https://github.com/docker/login-action#github-container-registry